### PR TITLE
CI configuration for openshift/kuryr-kubernetes

### PR DESCRIPTION
**Pull request by dulek**
Kuryr-Kubernetes is a formerly OpenStack project that uses OpenStack
Neutron to provide networking for Kubernetes pods. It increases
performance of OpenShift deployments that are running on an OpenStack
cloud.

This commit adds the CI configs for openshift/kuryr-kubernetes repo.

### CODE REVIEW

File: cluster/ci/config/prow/plugins.yaml [hunk 1]
```diff
@@ -248,6 +248,9 @@ plugins:
   openshift/knative-pkg:
   - approve
 
+  openshift/kuryr-kubernetes:
+    - approve
```
Comment (celebdor) [inline cluster/ci/config/prow/plugins.yaml]: why is it differently indented?

File: ci-operator/config/openshift/kuryr-kubernetes/openshift-kuryr-kubernetes-master.yaml [hunk 2]
```diff
@@ -0,0 +1,52 @@
+tag_specification:
+  name: '4.0'
```
Comment (petr-muller) [inline ci-operator/config/openshift/kuryr-kubernetes/openshift-kuryr-kubernetes-master.yaml]: Don't we now want `master` files to target `4.1`? @stevekuznetsov ?

Comment (stevekuznetsov) [reply ci-operator/config/openshift/kuryr-kubernetes/openshift-kuryr-kubernetes-master.yaml]: we do

Comment (dulek) [reply ci-operator/config/openshift/kuryr-kubernetes/openshift-kuryr-kubernetes-master.yaml]: Fixed, thanks!

File: cluster/ci/config/prow/plugins.yaml (hunk 1 above)
Comment (dulek) [reply cluster/ci/config/prow/plugins.yaml]: Done!

File: ci-operator/config/openshift/kuryr-kubernetes/openshift-kuryr-kubernetes-master.yaml [hunk 3]
```diff
@@ -0,0 +1,52 @@
+tag_specification:
+  name: '4.1'
```
Comment (petr-muller) [inline ci-operator/config/openshift/kuryr-kubernetes/openshift-kuryr-kubernetes-master.yaml]: Sorry about that, but the cut for 4.2 was done yesterday, so this should point to `4.2` now.

Comment (dulek) [reply ci-operator/config/openshift/kuryr-kubernetes/openshift-kuryr-kubernetes-master.yaml]: Not a problem, at least it solves my confusion with 4.1 being there when master is 4.2. ;)

**Pull request merged**